src/Sarif.Driver/Sdk/SkimmerBase.cs

```diff
@@ -7,17 +7,12 @@
 
 namespace Microsoft.CodeAnalysis.Sarif.Driver
 {
-    public abstract class SkimmerBase<TContext>  : PropertyBagHolder, ISkimmer<TContext>
+    public abstract class SkimmerBase<TContext>  : Rule, ISkimmer<TContext>
```

Comment:
>Rule [](start = 51, length = 4)

Rule derives from PropertyBag. The big problem with not using interfaces, of course, is that you need a single, sensible base class chain. We barely are able to accomplish this to eliminate IRule (thanks to the SARIF design change to make everything have a property bag).